tests/end-to-end/upgrade/pkg/tests/event-bus/event_bus.go

```diff
@@ -0,0 +1,397 @@
+	SubsInterface subscriptionClientSet.Interface
+}
+
+type eventBusFlow struct {
+	namespace string
+	log       logrus.FieldLogger
+	stop      <-chan struct{}
+
+	k8sInterface  kubernetes.Interface
+	eaInterface   eaClientSet.Interface
+	subsInterface subscriptionClientSet.Interface
+}
+
+// NewEventBusUpgradeTest returns new instance of the UpgradeTest
+func NewEventBusUpgradeTest(k8sCli kubernetes.Interface, eaInterface eaClientSet.Interface, subsCli subscriptionClientSet.Interface) *UpgradeTest {
+	return &UpgradeTest{
+		K8sInterface:  k8sCli,
+		EaInterface:   eaInterface,
+		SubsInterface: subsCli,
+	}
+}
+
+// CreateResources creates resources needed for e2e upgrade test
+func (ut *UpgradeTest) CreateResources(stop <-chan struct{}, log logrus.FieldLogger, namespace string) error {
+	return ut.newFlow(stop, log, namespace).createResources()
+}
+
+// TestResources tests resources after upgrade phase
+func (ut *UpgradeTest) TestResources(stop <-chan struct{}, log logrus.FieldLogger, namespace string) error {
+	return ut.newFlow(stop, log, namespace).testResources()
+}
+
+func (ut *UpgradeTest) newFlow(stop <-chan struct{}, log logrus.FieldLogger, namespace string) *eventBusFlow {
+	return &eventBusFlow{
+		log:       log,
+		stop:      stop,
+		namespace: namespace,
+
+		k8sInterface:  ut.K8sInterface,
+		eaInterface:   ut.EaInterface,
+		subsInterface: ut.SubsInterface,
+	}
+}
+
+func (f *eventBusFlow) createResources() error {
+	// iterate over steps
+	for _, fn := range []func() error{
+		f.createEventActivation,
+		f.createSubscription,
+		f.createSubscriber,
+		f.checkSubscriberStatus,
+		f.checkPublisherStatus,
+		f.checkSubscriptionReady,
+		f.publishTestEvent,
+		f.checkSubscriberReceivedEvent,
+	} {
+		err := fn()
+		if err != nil {
+			f.log.Errorf("CreateResources() failed with: %v", err)
+			return err
+		}
+	}
+	return nil
+}
+
+func (f *eventBusFlow) testResources() error {
+	// iterate over steps
+	for _, fn := range []func() error{
+		f.checkSubscriberStatus,
+		f.checkPublisherStatus,
+		f.checkSubscriptionReady,
+		f.publishTestEvent,
+		f.checkSubscriberReceivedEvent,
+		f.cleanup,
+	} {
+		err := fn()
+		if err != nil {
+			f.log.Errorf("TestResources() failed with: %v", err)
+			return err
+		}
+	}
+	return nil
+}
+
+func (f *eventBusFlow) createSubscriber() error {
+	if _, err := f.k8sInterface.AppsV1().Deployments(f.namespace).Get(subscriberName, metav1.GetOptions{}); err != nil {
+		f.log.Infof("Create Subscriber deployment")
+		if _, err := f.k8sInterface.AppsV1().Deployments(f.namespace).Create(util.NewSubscriberDeployment(subscriberImage)); err != nil {
+			f.log.Errorf("Create Subscriber deployment: %v\n", err)
+			return err
+		}
+		f.log.Infof("Create Subscriber service")
+		if _, err := f.k8sInterface.CoreV1().Services(f.namespace).Create(util.NewSubscriberService()); err != nil {
+			f.log.Errorf("Create Subscriber service failed: %v\n", err)
+		}
+		time.Sleep(30 * time.Second)
```

Comment:
In future you can consider using [function](https://github.com/kyma-project/kyma/blob/master/tests/end-to-end/upgrade/pkg/waiter/waiter.go) which already exist for waiting for long term actions